syntax/nix.vim

```diff
@@ -32,7 +32,7 @@ syn keyword nixTodo FIXME NOTE TODO OPTIMIZE XXX HACK contained
 syn match   nixComment '#.*' contains=nixTodo,@Spell
 syn region  nixComment start=+/\*+ end=+\*/+ contains=nixTodo,@Spell
 
-syn region nixInterpolation matchgroup=nixInterpolationDelimiter start="\${" end="}" contained contains=@nixExpr,nixInterpolationParam
+syn region nixInterpolation matchgroup=nixInterpolationDelimiter start="[^\$]\${" end="}" contained contains=@nixExpr,nixInterpolationParam
```

Comment:
skip might also work for start/end, otherwise use `\zs` to exclude the preceding character from matching.